datetime/src/main/java/com/fasterxml/jackson/datatype/jsr310/deser/DurationDeserializer.java

```diff
@@ -48,22 +58,35 @@ private DurationDeserializer()
     @Override
     public Duration deserialize(JsonParser parser, DeserializationContext context) throws IOException
     {
+        String string = parser.getText().trim();
         switch (parser.getCurrentTokenId())
         {
             case JsonTokenId.ID_NUMBER_FLOAT:
                 BigDecimal value = parser.getDecimalValue();
+                // If the decimal isnt within the bounds of a float, bail out
+                if(value.compareTo(INSTANT_MAX) > 0 ||
+                        value.compareTo(INSTANT_MIN) < 0) {
+                    NumberFormat formatter = new DecimalFormat("0.0E0");
+                    throw new JsonParseException(context.getParser(),
+                            String.format("Value of BigDecimal (%s) not within range to be converted to Duration",
+                                    formatter.format(value)));
+                }
+
                 long seconds = value.longValue();
                 int nanoseconds = DecimalUtils.extractNanosecondDecimal(value, seconds);
                 return Duration.ofSeconds(seconds, nanoseconds);
 
             case JsonTokenId.ID_NUMBER_INT:
+                if(string.length() > INSTANT_MAX_STRING_LEN) {
```

Comment:
Is the attack also possible via `BigInteger`, or just `BigDecimal`? I am thinking that (as per earlier comments by others) for common case there would be unnecessary allocation of another String, for the common case of input being relatively short.
And for special case of binary formats there would also be conversion from binary number into String.

So the extra checks would mostly make sense if use of `BigInteger` is also similarly dangerous.

Reply:
It's just `BigDecimal`, because `BigDecimal.longValue()` (and `intValue()` and `scale()` and some others) will consume unbounded resources even given short string representations of the decimal value.  It's the conversion to an integer that's the problem.

The input-string length checks in this PR are unrelated to [the reported DoS vector](https://github.com/FasterXML/jackson-databind/issues/2141).  IMO they should be submitted and considered separately, because they are muddying the analysis here by conflating independent concerns.

Reply:
I have removed all the string length checking entirely and it only contains the code to handle exponential numbers.